pkg/controller/build/build_controller.go

```diff
@@ -39,6 +39,27 @@ const succeedStatus string = "Succeeded"
 const namespace string = "namespace"
 const name string = "name"
 
+// error message
+const (
+	// validate secrets
+	ListSecretInNamespaceFailed string = "001: listing secrets in namespace %s failed"
+	NoSecretsInNamespace        string = "002: there are no secrets in namespace %s"
+	SecretsDoNotExist              string = "003: secrets %s do not exist"
+	SecretDoesNotExist             string = "004: secret %s does not exist"
+	// validate strategy
+	UnknownStrategy string = "005: unknown strategy %v"
```

Comment:
We still need a way to validate if error codes are modified or deleted, I quickly develop an example based in IOTA; please see https://play.golang.org/p/1td-qZ_HEbq

Reply:
Done!

Reply:
Hi @qu1queee ,

do we need to validate `if error codes are modified or deleted` in the `ValidateAndGenerateBuildErrorCode` function?

I think once there is any error code title or context changed, our build_controller unit test will notice it and fail the unit test, so that we can notice the change easily.

The current logic in the function is duplicate and I think no too much value for us:
```
	switch code {
	case ListSecretInNamespaceFailed:
		errorCode := "B" + string(code)
		return errorCode, nil
	case NoSecretsInNamespace:
		errorCode := "B" + string(code)
		return errorCode, nil
	case SecretsDoNotExist:
		errorCode := "B" + string(code)
		return errorCode, nil
	case SecretDoesNotExist:
		errorCode := "B" + string(code)
		return errorCode, nil
...
```

Reply:
yes, this is something I addressed with Zoe offline.